compilers/concrete-compiler/compiler/include/concretelang/Dialect/FHE/IR/FHEOps.td

```diff
@@ -455,6 +455,66 @@ def FHE_RoundEintOp: FHE_Op<"round", [Pure, UnaryEint, DeclareOpInterfaceMethods
     let hasFolder = 1;
 }
 
+def FHE_LsbEintOp: FHE_Op<"lsb", [Pure, UnaryEint, ConstantNoise]> {
+
+    let summary = "Extract the lowest significant bit at a given precision.";
+
+    let description = [{
+      This operation extract the lsb of a ciphertext in a specific precision.
+
+      Extracting only 1 bit:
+      ```mlir
+       // ok
+       %lsb = "FHE.lsb"(%a): (!FHE.eint<4>) -> (!FHE.eint<1>)
+
+      If you need to clear the lsb of the original ciphertext, you should extract to the same precision as the ciphertext.
+      If you need to extract several bits, you can extract sequentially using explicit bitwidth change and bit clearing.
+
+      Example:
+      ```mlir
+       // ok
+       %a_lsb = "FHE.lsb"(%a): (!FHE.eint<4>) -> (!FHE.eint<4>)
+       %a_lsb_cleared = "FHE.add_sub"(%a, %lsb) : (!FHE.eint<4>, !FHE.eint<4>) -> (!FHE.eint<4>)
+       %b = "FHE.reinterpret_precision(%a) : (!FHE.eint<4>) -> !FHE.eint<3>
+       // now you can extract the next lsb from %b
+       %b_lsb = "FHE.lsb"(%b): (!FHE.eint<3>) -> (!FHE.eint<3>)
+       // later if you need %b_lsb at the original position
+       %b_lsb_as_in_a = "FHE.reinterpret_precision(%b_lsb)" : (!FHE.eint<3>) -> !FHE.eint<4>
+    ```
+    }];
+
+    let arguments = (ins FHE_AnyEncryptedInteger:$input);
+    let results = (outs FHE_AnyEncryptedInteger);
+}
+
+def FHE_ReinterpretPrecisionEintOp: FHE_Op<"reinterpret_precision", [Pure, UnaryEint]> {
+
+    let summary = "Reinterpret the ciphertext with a different precision.";
+
+    let description = [{
+      It's a reinterpret cast changing only the precision.
+      On CRT represention, it does nothing.
```

Comment:
That's a bit weird to discuss about encoding and noise at this IR level, could we rewrite that?

Reply:
Ok, now at least native encoding is chosen, so there is no discussion.
I have removed the discussion on noise but mention correctness and small errors.